File: 800/703A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() 
{
    int t;
    cin >> t;

    int cnt_1 = 0, cnt_2 = 0;

    while (t--) 
    {
        int a, b;
        cin >> a >> b;

        if (a > b) 
            cnt_1++;
        if (a < b) 
            cnt_2++;
    }

    if (cnt_1 > cnt_2)
        cout << "Mishka" << endl;
    else if (cnt_1 < cnt_2)
        cout << "Chris" << endl;
    else
        cout << "Friendship is magic!^^" << endl;

    return 0;
}

File: 800/703A/main.java
import java.util.*;
public class main
{
    public static void main(String args[])
    {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        int cnt_1 = 0;
        int cnt_2 = 0;
        
        while(t-- > 0)
        {
            int a = sc.nextInt();
            int b = sc.nextInt();
            
            if(a > b)
                cnt_1++;
            if(a < b)
                cnt_2++;
        }
        
        if(cnt_1 > cnt_2)
            System.out.println("Mishka");
        else if(cnt_1 < cnt_2)
            System.out.println("Chris");
        else
            System.out.println("Friendship is magic!^^");
    }
}